### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cs-shubham-morankar**
#### Mantis:

- [0894467](https://mantis.fortinet.com/bug_view_page.php?bug_id=0894467) : Execute Query action does not work in connector Azure Log Analytics

#### UTCs:

- Verified Configuration.
- Verified all the operations.
- Verified all the playbooks(InActive state, Debug mode Off)

### CODE REVIEW

File: azure-log-analytics/microsoft_api_auth.py [hunk 1]
```diff
@@ -28,236 +29,106 @@
...
 class MicrosoftAuth:
-
     def __init__(self, config):
         self.client_id = config.get("client_id")
         self.client_secret = config.get("client_secret")
         self.verify_ssl = config.get('verify_ssl')
-        self.host = config.get("resource")
+        self.scope = config.get('scope', MANAGE_SCOPE)
+        self.host = "https://"
         if self.host[:7] == "http://":
```
Comment (cs-amol-phatangade) [inline azure-log-analytics/microsoft_api_auth.py]: If host is always use https then we don't need to check again that host is start from http or https.

File: azure-log-analytics/operations.py [hunk 2]
```diff
@@ -6,56 +6,82 @@
...
-def api_request(method, endpoint, connector_info, config, params=None, data=None, headers={}):
-    try:
-        ms = MicrosoftAuth(config)
-        token = ms.validate_token(config, connector_info)
-        headers['Authorization'] = token
-        headers['Content-Type'] = 'application/json'
-        logger.debug("Endpoint: {0}".format(endpoint))
+class AzureLogAnalytics(object):
+    def __init__(self, config):
+        self.server_url = 'https://api.loganalytics.io'
```
Comment (cs-amol-phatangade) [inline azure-log-analytics/operations.py]: Do we really need to hardcode this URL here or we can take this from config? We can set this URL as a default.

File: azure-log-analytics/microsoft_api_auth.py [hunk 3]
```diff
@@ -28,236 +29,106 @@
...
+    def validate_log_token(self, connector_config, connector_info):
+        ts_now = time()
+        expires = connector_config['logExpiresOn']
+        expires_ts = self.convert_ts_epoch(expires)
+        if ts_now > float(expires_ts):
+            logger.info("Token expired at {0}".format(expires))
+            token_resp = self.generate_token(LOG_SCOPE)
+            connector_config['logAccessToken'] = token_resp['accessToken']
+            connector_config['logExpiresOn'] = token_resp['expiresOn']
+            connector_config['logRefresh_token'] = token_resp.get('refresh_token')
```
Comment (cs-amol-phatangade) [inline azure-log-analytics/microsoft_api_auth.py]: Why this key is not in camelcase format? is there any requirement we need this key like this? if not then try to maintain all keys either camelcase or snake_case.

**Review by cs-amol-phatangade — APPROVED** [review]

File: azure-log-analytics/operations.py (hunk 2 above)
Comment (cs-shubham-morankar) [reply azure-log-analytics/operations.py]: We are using two base url to execute this connector actions.

https://management.azure.com -> Saved Search actions
https://api.loganalytics.io -> Execute Query action

We don't have any default url for this connector.

File: azure-log-analytics/microsoft_api_auth.py (hunk 1 above)
Comment (cs-shubham-morankar) [reply azure-log-analytics/microsoft_api_auth.py]: We are removing all this lines

self.host = "https://"
if self.host[:7] == "http://":
       self.host = self.host.replace('http://', 'https://')
elif self.host[:8] == "https://":
       self.host = "{0}".format(self.host)
else:
       self.host = "https://{0}".format(self.host)

No use of this lines in the code. Thanks for pointing out.

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**